src/hostapi/pulseaudio/pa_linux_pulseaudio_cb.c

```diff
@@ -420,6 +440,26 @@ static int _PaPulseAudio_ProcessAudio(PaPulseAudio_Stream *stream,
             PaUtil_SetOutputFrameCount( &stream->bufferProcessor,
                                         hostFramesPerBuffer );
 
+        }
+
+        hostFrameCount =
+            PaUtil_EndBufferProcessing( &stream->bufferProcessor,
+                                        &ret );
+
+        PaUtil_EndCpuLoadMeasurement( &stream->cpuLoadMeasurer,
+                                      hostFrameCount );
+
```

Comment:
I think you should move the `return paNotInitialized` statement here, so that you exit with the buffer processing in a known good state. Obviously this is non-trivial to do properly.

Reply:
I agree. Returning from the middle of a begin/end block seems wrong. Best to cleanup before returning.

Reply:
ok this one should be now corrected